numpy/core/tests/test_records.py

```diff
@@ -325,6 +326,24 @@ def test_zero_width_strings(self):
         assert_equal(rec['f1'], [b'', b'', b''])
 
 
+@pytest.mark.skipif(Path is None, reason="No pathlib.Path")
+class TestPathUsage(object):
+    # Test that pathlib.Path can be used
+    def test_tofile_fromfile(self):
+        with temppath(suffix='.bin') as path:
+            path = Path(path)
+            a = np.empty(10, dtype='f8,i4,a5')
+            a[5] = (0.5,10,'abcde')
+            a.newbyteorder('<')
+            with path.open("wb") as fd:
+                a.tofile(fd)
```

Comment:
`ndarray.tofile()` [discards information on endianness](https://docs.scipy.org/doc/numpy-1.15.0/reference/generated/numpy.ndarray.tofile.html) and most recent ARM features [switchable endianness (bi-endian)](https://docs.python.org/3/library/struct.html)

I wonder if this is contributing to #12330.

Certainly this test has just started failing stochastically on shippable / ARM for 2.7 / 3.7, so it may not have been stable in the first place given how recently this was merged.

Reply:
This wasn't the problem -- I'm preparing the patch now in the linked PR, which is related to testing precision and np.empty() as far as I can tell.